tests/foreman/destructive/test_installer.py

```diff
@@ -17,13 +17,29 @@
 :Upstream: No
 """
 import pytest
+from fauxfactory import gen_domain
+from fauxfactory import gen_string
 
 from robottelo.config import settings
 from robottelo.utils.installer import InstallerCommand
 
 pytestmark = pytest.mark.destructive
 
 
+@pytest.fixture
+def set_random_fqdn(target_sat, request):
+    shortname = gen_string('alpha')
+    new_domain = gen_domain()
+    target_sat.execute(
+        f'echo "search {new_domain}" >> /etc/resolv.conf; hostnamectl set-hostname {shortname}'
+    )
+    yield shortname, new_domain
+
+    @request.addfinalizer
+    def _finalize():
+        target_sat.execute(f'hostnamectl set-hostname {target_sat.hostname}')
```

Comment:
If this is for tear down, you should be able to just put:
```suggestion
    yield shortname, new_domain
    target_sat.execute(f'hostnamectl set-hostname {target_sat.hostname}')
```
Everything after the yield will run once the test is done.

Reply:
oh right, I forgot. thanks for noticing it.

Reply:
updated the pr.